### PR TITLE
created batch file for contact processing with systemNotificationCont…

### PR DESCRIPTION
**Pull request by lynnmichels**
created batch file for contact processing with systemNotificationController

### CODE REVIEW

File: src/classes/ContactBatch.cls [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	}//end execute method
+    
+    global void finish(Database.BatchableContext info){   
+       Outage_Log__c outageSettings = Outage_Log__c.getOrgDefaults();
+       //Check batch status - process or send error email
+       AsyncApexJob job = [SELECT Id, Status, ExtendedStatus
+        				 FROM AsyncApexJob
+                         WHERE Id =: info.getJobId()];        
+       if(job.Status == 'Completed'){
+           if(currentOutage){
```
Comment (Qu1kee) [inline src/classes/ContactBatch.cls]: could refactor to: updateOutageSetting(String status)

File: src/classes/ContactBatch.cls [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            }
+            contactsToUpdate.add(c);
+        }//end for
+		update contactsToUpdate;
+	}//end execute method
+    
+    global void finish(Database.BatchableContext info){   
+       Outage_Log__c outageSettings = Outage_Log__c.getOrgDefaults();
+       //Check batch status - process or send error email
+       AsyncApexJob job = [SELECT Id, Status, ExtendedStatus
```
Comment (Qu1kee) [inline src/classes/ContactBatch.cls]: SOQL as separate function

File: src/classes/ContactBatchTest.cls [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    @isTest static void contactBatch_executeBatch_notifyContactsOfOutage_successfulBatch() {
+        //create custom setting with no values
+		Outage_Log__c ol = new Outage_Log__c();
+		insert ol;
+        
+    	List<Contact> contactList = new List<Contact>();
+		Account acct = new Account(Name = 'testAccount');
+		insert acct;
+
+		Contact c1 = new Contact (LastName = 'testContact1', AccountId = acct.Id, HixnyRole_del__c= 'System User');
```
Comment (Qu1kee) [inline src/classes/ContactBatchTest.cls]: refactor for both test functions to use: createContacts(Boolean outage)

File: src/classes/ContactBatchTest.cls [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+		contactList.add(c1);
+		Contact c2 = new Contact (LastName = 'testContact2', AccountId = acct.Id, HixnyRole_del__c= 'Trusted Agent,System User');
+		contactList.add(c2);
+		Contact c3 = new Contact (LastName = 'testContact3', AccountId = acct.Id, HixnyRole_del__c= '');
+		contactList.add(c3);
+		insert contactList;
+        //currently no outage and no batch running
+        system.assertEquals(ol.System_outage__c, false);
```
Comment (Qu1kee) [inline src/classes/ContactBatchTest.cls]: assertEquals(expected, actual).  You have it switched

File: src/classes/ContactBatch.cls [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    public ContactBatch(boolean crtOutage){
+        currentOutage = crtOutage;
+    }
+    
+    global Database.QueryLocator start(Database.BatchableContext info){ 
+       if(currentOutage){
+           return Database.getQueryLocator('SELECT Id FROM Contact WHERE Outage_Notification__c = true');
+       }
+       else{
+       		return Database.getQueryLocator('SELECT Id FROM Contact WHERE HixnyRole_del__c LIKE \'%System User%\'');
```
Comment (shanekenyon87) [inline src/classes/ContactBatch.cls]: What happens if no records are found?

File: src/classes/ContactBatch.cls [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+       }
+    }//end start method     
+    
+    global void execute(Database.BatchableContext BC, List<Contact> scope){
+	List<Contact> contactsToUpdate = new List<Contact>();    	
+        for(Contact c : scope){
```
Comment (shanekenyon87) [inline src/classes/ContactBatch.cls]: Please use the formatting utility in Vscode.  for instance `for (Contact c : scope) {` is what we'd want.  Right-click into a file and choose "Format Document" - not sure if we need to ensure the right linter is installed.  But in any case, this is what we are hewing to: https://google.github.io/styleguide/javaguide.html

File: src/classes/ContactBatch.cls [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+            if (currentOutage){
+            	c.Outage_Notification__c = false;
+            }
+            else{
+            	c.Outage_Notification__c = true;   
+            }
```
Comment (shanekenyon87) [inline src/classes/ContactBatch.cls]: This could be written `c.Outage_Notification__c = !currentOutage;`

File: src/classes/ContactBatch.cls [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+           update outageSettings;
+       }//end if Completed
+       else if(job.Status == 'Failed'){
+           sendErrorEmail(job.id, job.Status, job.ExtendedStatus);
+       }//end if Failed or Aborted    
+   }//end finish method
+    
+	public void sendErrorEmail(id batchJobId, string batchJobStatus, string batchJobExtendedStatus){	
+		Messaging.SingleEmailMessage mail = new Messaging.SingleEmailMessage();
+		String[] toAddresses = new String[] {'salesforce+hixny@redargyle.com'};
```
Comment (shanekenyon87) [inline src/classes/ContactBatch.cls]: Ideally we should load email addresses from config

**Pull request closed without merging**